### PR TITLE
dist: make ng_sniffer.py more pythonic

### PR DESCRIPTION
**Pull request by authmillenon**
Depends on #3150.

### CODE REVIEW

**Comment by haukepetersen** [comment]
nice. While you are at it, could we also include some lines in the configuration part, that check if the device was able to set the options `RAW`, `PROMISC` and the channel? -> but makes probably sense in a new PR...

**Comment by authmillenon** [comment]
Yes, I can make this in a follow-up PR.

**Comment by authmillenon** [comment]
Just a remark: I wanted to make it version-independent, too. But `pack` works a little different in python3 (returning `bytes` instead of `str`) which requires me to encode the result as string to put it out. Need to investigate first what would be the appropriate output format for that.

**Comment by gebart** [comment]
@authmillenon 
>But pack works a little different in python3 (returning bytes instead of str) which requires me to encode the result as string to put it out.

I don't understand what you mean, does it not work in python3 right now?

I use pack all the time in python3 for writing to file using outfile.write(bytes) and I don't remember having to do anything special to get it working..

**Comment by authmillenon** [comment]
@gebart I don't really know… my interpreter was complaining about writing bytes to outfile. I'll try again when I find the time. pyserial was also a little bit cranky regarding this.

**Comment by authmillenon** [comment]
@gebart when I fix for python3 until this point I get

```
$ python3 ./dist/tools/ng_sniffer/ng_sniffer.py /dev/ttyUSB1 500000 17
ifconfig 4 set chan 17
ifconfig 4 raw
ifconfig 4 promisc
Traceback (most recent call last):
  File "./dist/tools/ng_sniffer/ng_sniffer.py", line 139, in <module>
    main(sys.argv)
  File "./dist/tools/ng_sniffer/ng_sniffer.py", line 132, in main
    generate_pcap(serport, outfile)
  File "./dist/tools/ng_sniffer/ng_sniffer.py", line 79, in generate_pcap
    NETWORK))
TypeError: must be str, not bytes
```

on the `write.out()`.

My changes so far:

```diff
diff --git a/dist/tools/ng_sniffer/ng_sniffer.py b/dist/tools/ng_sniffer/ng_sniffer.py
index 2ec82fc..b4fc2c5 100755
--- a/dist/tools/ng_sniffer/ng_sniffer.py
+++ b/dist/tools/ng_sniffer/ng_sniffer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
 '''
 (C) 2012, Mariano Alvira <mar@devl.org>
 (C) 2014, Oliver Hahm <oliver.hahm@inria.fr>
@@ -30,6 +30,7 @@ OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 SUCH DAMAGE.
 '''
 
+from __future__ import print_function
 import sys
 import re
 from time import sleep, time
@@ -49,24 +50,25 @@ NETWORK = 230       # 802.15.4 no FCS
 def configure_interface(port, channel):
     line = ""
     iface = 0
-    port.write('ifconfig\n')
+    port.write(b'ifconfig\n')
     while True:
         line = port.readline()
         if line == '':
-            print >> sys.stderr, "Application has no network interface defined"
+            print("Application has no network interface defined",
+                  file=sys.stderr)
             sys.exit(2)
-        match = re.search(r'^Iface +(\d+)', line)
+        match = re.search(r'^Iface +(\d+)', line.decode())
         if match is not None:
             iface = int(match.group(1))
             break
 
     # set channel, raw mode, and promiscuous mode
-    print >> sys.stderr, 'ifconfig %d set chan %d' % (iface, channel)
-    print >> sys.stderr, 'ifconfig %d raw' % iface
-    print >> sys.stderr, 'ifconfig %d promisc' % iface
-    port.write('ifconfig %d set chan %d\n' % (iface, channel))
-    port.write('ifconfig %d raw\n' % iface)
-    port.write('ifconfig %d promisc\n' % iface)
+    print('ifconfig %d set chan %d' % (iface, channel), file=sys.stderr)
+    print('ifconfig %d raw' % iface, file=sys.stderr)
+    print('ifconfig %d promisc' % iface, file=sys.stderr)
+    port.write(('ifconfig %d set chan %d\n' % (iface, channel)).encode())
+    port.write(('ifconfig %d raw\n' % iface).encode())
+    port.write(('ifconfig %d promisc\n' % iface).encode())
 
 
 def generate_pcap(port, out):
@@ -102,9 +104,9 @@ def generate_pcap(port, out):
 
 def main(argv):
     if len(argv) < 4:
-        print >> sys.stderr, "Usage: %s tty baudrate channel [outfile]" % \
-            (argv[0])
-        print >> sys.stderr, "       channel = 11-26"
+        print("Usage: %s tty baudrate channel [outfile]" % (argv[0]),
+              file=sys.stderr)
+        print("       channel = 11-26", file=sys.stderr)
         sys.exit(2)
 
     # open serial port
@@ -112,7 +114,7 @@ def main(argv):
         serport = Serial(argv[1], argv[2], dsrdtr=0, rtscts=0,
                          timeout=1)
     except IOError:
-        print >> sys.stderr, "error opening port"
+        print("error opening port", file=sys.stderr)
         sys.exit(2)
 
     sleep(1)
@@ -129,7 +131,7 @@ def main(argv):
     try:
         generate_pcap(serport, outfile)
     except KeyboardInterrupt:
-        print ""
+        print()
         sys.exit(2)
 
 
```

**Comment by authmillenon** [comment]
```
$ python3 --version
Python 3.4.0
```

**Comment by authmillenon** [comment]
Got it… `stdout` for binary data ;-)

**Comment by authmillenon** [comment]
Rebased to current master.

**Comment by authmillenon** [comment]
Reassigned to take some load of @haukepetersen.

File: dist/tools/ng_sniffer/ng_sniffer.py [hunk 1]
```diff
@@ -90,38 +44,100 @@
...
+
+        pkt_header = re.match(r">? *rftest-rx --- len 0x(\w\w).*",
+                              line.decode())
         if pkt_header:
-            t = time.time()
-            sec = int(t)
-            usec = (t - sec) * 100000
+            now = time()
+            sec = int(now)
+            usec = (int(now) - sec) * 100000
```
Comment (gebart) [inline dist/tools/ng_sniffer/ng_sniffer.py]: This doesn't make alot of sense, does it? `int(now) - int(now)` will always be zero

File: dist/tools/ng_sniffer/ng_sniffer.py [hunk 2]
```diff
@@ -90,38 +44,100 @@
...
+
+    # open serial port
+    try:
+        serport = Serial(argv[1], argv[2], dsrdtr=0, rtscts=0,
+                         timeout=1)
+    except IOError:
+        print("error opening port", file=sys.stderr)
+        sys.exit(2)
+
+    sleep(1)
```
Comment (gebart) [inline dist/tools/ng_sniffer/ng_sniffer.py]: remove the sleep if not needed, otherwise add a comment in the code why it is necessary to sleep.

**Comment by gebart** [comment]
As a follow up PR I would suggest adding argparse to handle the command line arguments.

File: dist/tools/ng_sniffer/ng_sniffer.py (hunk 2 above)
Comment (authmillenon) [inline dist/tools/ng_sniffer/ng_sniffer.py]: I'm just restructuring code here. I have no idea if it is needed (but I would guess so) or why. It certain may does not hurt to wait for the reaction to the input.

File: dist/tools/ng_sniffer/ng_sniffer.py (hunk 1 above)
Comment (authmillenon) [inline dist/tools/ng_sniffer/ng_sniffer.py]: True. Will fix.

Comment (authmillenon) [inline dist/tools/ng_sniffer/ng_sniffer.py]: Fixed

**Comment by authmillenon** [comment]
@gebart ping?

**Comment by gebart** [comment]
ACK

**Comment by gebart** [comment]
please squash

**Comment by authmillenon** [comment]
Squashed

**Comment by PeterKietzmann** [comment]
I see ACKs and green lights from all sides. And go

File: dist/tools/ng_sniffer/ng_sniffer.py (hunk 1 above)
Comment (gebart) [inline dist/tools/ng_sniffer/ng_sniffer.py]: This fix was either never pushed before merging, or accidentally reverted during a rebase...

**Pull request merged**